plugins/chain/plugin.cpp

```diff
@@ -95,6 +98,25 @@ namespace golos { namespace plugins { namespace chain {
             ++itr;
             db.remove(vote);
         }
+
+        if (db.has_hardfork(STEEMIT_HARDFORK_0_21__1013)) {
+            const auto& mprops = db.get_witness_schedule_object().median_props;
+
+            if (db.clear_old_worker_techspec_approves()) {
+                const auto& wtao_idx = db.get_index<golos::chain::worker_techspec_approve_index, golos::chain::by_techspec_approver>();
+
+                const auto& idx = db.get_index<golos::chain::worker_techspec_index>().indices().get<golos::chain::by_created>();
+                auto itr = idx.begin();
+                while (itr != idx.end() && itr->created + mprops.worker_techspec_approve_term_sec <= db.head_block_time()) {
```

Comment:
You should close worker tspec in this place.
And only for such tspecs you should remove the approves, because they are not more needed.